subprojects/core-api/src/main/java/org/gradle/api/Task.java

```diff
@@ -181,6 +181,8 @@
     /**
      * <p>Returns the name of this task. The name uniquely identifies the task within its {@link Project}.</p>
      *
+     * <p>This method should never be implemented by a custom Task.</p>
```

Comment:
This is true for all methods on this interface. Maybe we should move it to the class Javadoc?

Reply:
Makes sense, I will move to the class Javadoc